tests/main/presenters/test_search_presenters.py

```diff
@@ -194,26 +195,52 @@ def test_request_filters_are_set(self):
     def test_show_lots_and_categories_selection(self):
         framework = self._get_framework_fixture_data('g-cloud-9')['frameworks']
         lots = framework['lots']
-        lot_slug = 'cloud-software'
-
-        category_filter_group = filters_for_lot(lot_slug, g9_builder)['categories-example']
-
-        with self.app.test_request_context("/g-cloud/search?q=&lot={}".format(lot_slug)):
-            annotate_lots_with_categories_selection(lots, category_filter_group, flask.request)
-
-            assert lots[0].get('slug') == 'cloud-hosting'
-            assert not lots[0].get('selected')
-            assert not lots[0].get('categories')
-
-            assert lots[1].get('slug') == 'cloud-software'
-            assert lots[1].get('selected')
-            category_links = lots[1]['categories']
-            assert 'checkboxTreeExample=option+1' in category_links[0]['link']
-            assert not category_links[0].get('children')
-            assert 'checkboxTreeExample=option+2' in category_links[1]['link']
-            sub_categories = category_links[1]['children']
-            assert 'checkboxTreeExample=option+2.1' in sub_categories[0]['link']
-            assert 'checkboxTreeExample=option+2.2' in sub_categories[1]['link']
+
+        category_filter_group = filters_for_lot('cloud-software', g9_builder)['categories-example']
+        # in this test, the key 'category' key is 'checkboxTreeExample'
+
+        # first test with a top-level category selected
+        url = "/g-cloud/search?q=&lot=cloud-software&otherfilter=somevalue&checkboxTreeExample=option+1"
+        with self.app.test_request_context(url):
+            tree_root = build_lots_and_categories_link_tree(lots, category_filter_group, flask.request)
+
+            assert tree_root.get('name') == 'All categories'
+
+            lot_filters = tree_root['children']
+            selected_lot = next(f for f in lot_filters if f['selected'])
+            assert selected_lot.get('name') == 'Cloud software'
+            # there should be a link to the lot without a category, as a category has been selected within it
+            assert 'lot={}'.format('cloud-software') in selected_lot['link']
+            assert 'checkboxTreeExample' not in selected_lot['link']
+
+            # check that we have links in place to a search with the relevant category filter applied,
+            # except to the currently-selected category
+            category_filters = selected_lot['children']
+            assert 'link' not in category_filters[0]
+            assert not category_filters[0].get('children')
```

Comment:
Can we do the same as the line above to be consistent/pin this more precisely? Eg `'children' not in ...`. Or, if it should be there but empty, `== []/is None`. Quite a few things could be Falsey.